src/barfturn.ts

```diff
@@ -0,0 +1,270 @@
+import {
+  availableAmount,
+  canAdventure,
+  currentRound,
+  inebrietyLimit,
+  myAdventures,
+  myInebriety,
+  myLevel,
+  runChoice,
+  runCombat,
+  totalTurnsPlayed,
+  toUrl,
+  use,
+  useFamiliar,
+  visitUrl,
+} from "kolmafia";
+import {
+  $effect,
+  $item,
+  $items,
+  $location,
+  $monster,
+  $skill,
+  adventureMacro,
+  adventureMacroAuto,
+  clamp,
+  Counter,
+  ensureEffect,
+  get,
+  getModifier,
+  have,
+  Requirement,
+  SourceTerminal,
+} from "libram";
+import { Macro, withMacro } from "./combat";
+import { completeBarfQuest } from "./dailies";
+import { estimatedTurns } from "./embezzler";
+import { barfFamiliar, freeFightFamiliar, meatFamiliar } from "./familiar";
+import {
+  embezzlerLog,
+  globalOptions,
+  kramcoGuaranteed,
+  safeRestore,
+  setChoice,
+  steveAdventures,
+} from "./lib";
+import { meatMood } from "./mood";
+import {
+  familiarWaterBreathingEquipment,
+  freeFightOutfit,
+  meatOutfit,
+  tryFillLatte,
+  waterBreathingEquipment,
+} from "./outfit";
+import { determineDraggableZoneAndEnsureAccess, digitizedMonstersRemaining } from "./wanderer";
+const embezzler = $monster`Knob Goblin Embezzler`;
+
+function embezzlerPrep(requirements?: Requirement) {
+  useFamiliar(meatFamiliar());
+  meatOutfit(true, requirements);
+}
+
+function freeFightPrep(requirements?: Requirement) {
+  useFamiliar(freeFightFamiliar(true));
+  freeFightOutfit(requirements);
+}
+
+const sober = () => myInebriety() <= inebrietyLimit();
+
+type Turn = {
+  name: string;
+  available: () => boolean;
+  perform: () => boolean;
+};
+
+const turns: Turn[] = [
+  {
+    name: "Lights Out",
+    available: () =>
+      totalTurnsPlayed() % 37 === 0 && totalTurnsPlayed() !== get("lastLightsOutTurn"),
+    perform: () => {
+      const steveRoom = get("nextSpookyravenStephenRoom");
+      const ghostLocation = get("ghostLocation");
+      if (steveRoom && canAdventure(steveRoom) && steveRoom !== ghostLocation) {
+        const fightingSteve = steveRoom === $location`The Haunted Laboratory`;
+        // Technically drops 500 meat, but that's close enough for me.
+        if (fightingSteve) embezzlerPrep();
+        const plan = steveAdventures.get(steveRoom);
+        if (plan) {
+          withMacro(
+            Macro.if_($monster`Stephen Spookyraven`, Macro.basicCombat()).abort(),
+            () => {
+              visitUrl(toUrl(steveRoom));
+              for (const choiceValue of plan) {
```

Comment:
i am very confused about what plan is doing here and why you're accessing it. what's the point of "plan" rather than just runChoice within haunted laboratory. maybe this would make more sense with better var names?

Reply:
We need the plan because many of these adventures will lead through several phases of the choice without changing choiceID, because the game is dumb.

It's essentially a "plan" for what choice options we're going to submit in what order.